code/game/gamemodes/umbrage/umbrage_objects.dm

```diff
@@ -13,10 +13,12 @@
 /obj/item/weapon/dark_bead/New()
 	..()
 	animate(src, alpha = 30, time = 30)
-	spawn(30)
-		if(!eating)
-			loc << "<span class='warning'>You were too slow! [src] faded away.</span>"
-			qdel(src)
+	QDEL_IN(src, 30)
+
+/obj/item/weapon/dark_bead/Destroy()
+	if(isliving(loc) && !eating)
```

Comment:
The QDEL_IN timer isn't removed when you start eating; you need to do var/timerid on the bead, then timerid = QDEL_IN() and deltimer(timerid) where appropriate